src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void save(TaskList tasks) {
+            File f = new File(filePath);
+            if (!f.exists()) {
+                try {
+                    f.createNewFile();
+                    FileWriter fw = new FileWriter(filePath);
+                    for (int i = 0; i < tasks.getLength(); i++) {
+                        fw.write(tasks.getTask(i).savedString() + System.lineSeparator());
+                    }
+                    fw.close();
+                } catch (IOException e) {
+                    Ui.printCannotOpenFile();
+                }
+            } else {
+                try {
+                    FileWriter fw = new FileWriter(filePath);
+                    for (int i = 0; i < tasks.getLength(); i++) {
+                        fw.write(tasks.getTask(i).savedString() + System.lineSeparator());
+                    }
+                    fw.close();
+                } catch (IOException e) {
+                    Ui.printUnableToAppend();
+                }
+            }
+    }
```

Comment:
Perhaps the duplicate code could be separated into a method instead? I noticed some duplicate code in other places too.

Reply:
Oo, okay, I'll try that 👍